src/main/java/uk/gov/hmcts/divorce/common/event/SwitchedToSoleCo.java

```diff
@@ -89,15 +89,15 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
         if (ccdAccessService.isApplicant2(httpServletRequest.getHeader(AUTHORIZATION), caseId)) {
             log.info("Request made by applicant to switch to sole for case id: {}", caseId);
             switchToSoleService.switchApplicantData(data);
-            switchToSoleService.switchUserRoles(data, caseId);
+            switchToSoleService.switchUserRoles(beforeDetails.getData(), caseId);
```

Comment:
I think this switchUserRoles is called in other places, do we need a task to check it's called with the correct data for those?